src/client/components/hud/Initiative.tsx

```diff
@@ -252,17 +259,82 @@ export function InitiativeHUD() {
 
   const focusTokenOnTurnStart = useRRSettings()[0].focusTokenOnTurnStart;
 
+  const updateConcentration = useCallback(
+    (
+      character: RRCharacter,
+      updater: React.SetStateAction<RRCharacter["currentlyConcentratingOn"]>
+    ) => {
+      dispatch((state) => {
+        const old =
+          state.characters.entities[character.id]?.currentlyConcentratingOn;
+
+        if (old === undefined) return [];
+        const newConcentration =
+          typeof updater === "function" ? updater(old) : updater;
+
+        return {
+          actions: [
+            characterUpdate({
+              id: character.id,
+              changes: { currentlyConcentratingOn: newConcentration },
+            }),
+          ],
+          optimisticKey: "concentration",
+          syncToServerThrottle: 0,
+        };
+      });
+    },
+    [dispatch]
+  );
```

Comment:
We might not even need this code (see my comment below), but even if we do, this could be simplified since we no longer use the function version of the `updater`, but just the version where we pass a fixed value (`null`).